### ISSUE TITLE
MaxDB (the DBMS of SAP) support

### ISSUE THREAD

**Issue by alexanderadam** [johnjansen/crystal-libraries-needed#100]
I'm not sure whether it should be here or as an issue on the [crystal-db](https://github.com/crystal-lang/crystal-db) project but it would be nice to be able to connect to [MaxDB](https://db.apache.org/ddlutils/databases/maxdb.html). The [Ruby gem is using the JDBC driver](https://github.com/SAP/cloud-jdbc-maxdb-gem) for that.

**Comment by RX14** [comment]
That ruby gem only works on jruby by loading java code. If nobody has managed to port it to ruby, let alone crystal, I doubt this will happen any time soon.

**Comment by alexanderadam** [comment]
Yes, the gem was written by SAP so I guess they wanted to provide compatibility to their database without investing too much resources into Ruby.
On the other hand there's even a driver for [PHP](http://svn.php.net/viewvc/pecl/maxdb/trunk/) that is mostly written in C (see [`php_maxdb.c`](http://svn.php.net/viewvc/pecl/maxdb/trunk/php_maxdb.c?view=markup) and [`php_maxdb.h`](http://svn.php.net/viewvc/pecl/maxdb/trunk/php_maxdb.h?view=markup)). And because Crystal and C are very good friends I wont loose any hope. :wink: 

But @RX14, independent from that, do you think the place for this issue here is fine or should I rather move this to the crystal-db project? I guess it should be fine here because #70 is here as well?

**Comment by RX14** [comment]
It belongs here. If there's a C driver, then that's good! It can probably be bound.

**Comment by alexanderadam** [comment]
For anyone who wants to have a look at it:
Here is the PHP C driver (just in case anything happens with the linked svn repository).

[php_maxdb.c.txt](https://github.com/crystal-community/crystal-libraries-needed/files/2723947/php_maxdb.c.txt)
[php_maxdb.h.txt](https://github.com/crystal-community/crystal-libraries-needed/files/2723948/php_maxdb.h.txt)

There are some calls that create PHP functions (i.e. `PHP_FUNCTION`, `PHP_FALIAS`, `ZEND_DECLARE_MODULE_GLOBALS` and `zend_function_entry`) and I guess those have to be rewritten / removed / replaced by Crystal defintions.

**Comment by konovod** [comment]
PHP driver is basically a wrapper around a C SDK: http://maxdb.sap.com/documentation/sqldbc/SQLDBC_API
and there is even an .h file available without registration - http://maxdb.sap.com/documentation/sqldbc/SQLDBC_API/SQLDBC__C_8h-source.html

**Comment by alexanderadam** [comment]
@konovod you are clearly better in googling than me :wink: :+1: 

Seems also to be available [on GitHub](https://github.com/gitpan/DBD-MaxDB).
